### PR TITLE
New feature to measure the latency introduced by the CDS

### PR DESCRIPTION
**Pull request by jvelazquezm**

### CODE REVIEW

File: kubernetes/crypto_detector/consumer.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        - containerPort: 12002
+        args: ["-b", "kafka:9092", "-c", "salida"]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: consumer-service
+  labels:
+    name: consumer-app
+    id: "2"
```
Comment (idomingu) [inline kubernetes/crypto_detector/consumer.yaml]: Do we need these IDs?

File: kubernetes/crypto_detector/producer.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        - containerPort: 12001
+        args: ["-b", "kafka:9092", "-p", "entrada", "-t", "1"]
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: producer-service
+  labels:
+    name: producer-app
+    id: "1"
```
Comment (idomingu) [inline kubernetes/crypto_detector/producer.yaml]: Same here

File: kubernetes/crypto_detector/stats/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+```
+Change n with the number of seconds that the script will be collecting statistics.
+
+To delete the scenario after being deployed:
+```
+./clean-env.sh
+```
+
+### Important notes:
+- The script creates a local cluster with Zookeeper and Kafka in Step 1. Comment these lines if not needed
```
Comment (idomingu) [inline kubernetes/crypto_detector/stats/README.md:19]: Note that Helm is required in this step.

File: docker/crypto-detector/producer/producer.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    KAFKA_BROKER = args["broker"]
+    KAFKA_TOPIC_PRODUCE = args["produce"]
+    TIME_BETWEEN_PRODUCE = args["time"]
+
+    benign_traffic = 'ts,te,td,sa,da,sp,dp,pr,flg,fwd,stos,ipkt,ibyt,opkt,obyt,in,out,sas,das,smk,dmk,dtos,dir,nh,nhb,svln,dvln,ismc,odmc,idmc,osmc,mpls1,mpls2,mpls3,mpls4,mpls5,mpls6,mpls7,mpls8,mpls9,mpls10,cl,sl,al,ra,eng,exid,tr,tpkt,tbyt,cp,prtcp,prudp,pricmp,prigmp,prother,flga,flgs,flgf,flgr,flgp,flgu,64.95464773702646,74.23388312803024,32704.665135592822,3164.219268332289,503.5,42.625,10.335777126099709,0.875'
+    crypto_traffic = 'ts,te,td,sa,da,sp,dp,pr,flg,fwd,stos,ipkt,ibyt,opkt,obyt,in,out,sas,das,smk,dmk,dtos,dir,nh,nhb,svln,dvln,ismc,odmc,idmc,osmc,mpls1,mpls2,mpls3,mpls4,mpls5,mpls6,mpls7,mpls8,mpls9,mpls10,cl,sl,al,ra,eng,exid,tr,tpkt,tbyt,cp,prtcp,prudp,pricmp,prigmp,prother,flga,flgs,flgf,flgr,flgp,flgu,31.94037796113921,31.94037796113921,14213.468192706949,3119.510247537929,445.0,97.66666666666669,4.5563139931740615,1.0'
+    producer = Producer()
+
+    print("Passed: -b " + KAFKA_BROKER + " -p " + KAFKA_TOPIC_PRODUCE + " -t " + TIME_BETWEEN_PRODUCE)
```
Comment (idomingu) [inline docker/crypto-detector/producer/producer.py]: Use logging as you did above.

File: kubernetes/crypto_detector/stats/stats.csv [hunk 5]
```diff
@@ -0,0 +1 @@
+
```
Comment (idomingu) [inline kubernetes/crypto_detector/stats/stats.csv]: Do we need this empty file? Not a good idea.

File: docker/crypto-detector/consumer/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
+FROM ubuntu:22.04
```
Comment (idomingu) [inline docker/crypto-detector/consumer/Dockerfile]: There are Docker images specifically prepared for Python. Check other examples in our repo like (ignore the rest of contents of the Dockerfile)  https://github.com/giros-dit/semantic-data-aggregator/blob/2c5fcc7df37add1284c0cc0946109a3f5fbce6c7/docker/data-lake-explorer/Dockerfile.

File: docker/crypto-detector/consumer/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+RUN apt-get update && apt-get install -y \
```
Comment (idomingu) [inline docker/crypto-detector/consumer/Dockerfile]: You won't need this command if you use the Docker image that ships python.

**Review by idomingu — CHANGES_REQUESTED** [review]

File: kubernetes/crypto_detector/consumer.yaml (hunk 1 above)
Comment (jvelazquezm) [reply kubernetes/crypto_detector/consumer.yaml]: Id deleted, not needed

File: kubernetes/crypto_detector/producer.yaml (hunk 2 above)
Comment (jvelazquezm) [reply kubernetes/crypto_detector/producer.yaml]: Id deleted, not needed

File: kubernetes/crypto_detector/stats/README.md (hunk 3 above)
Comment (jvelazquezm) [reply kubernetes/crypto_detector/stats/README.md:19]: Included a list of requirements to clarify

File: kubernetes/crypto_detector/stats/stats.csv (hunk 5 above)
Comment (jvelazquezm) [reply kubernetes/crypto_detector/stats/stats.csv]: Now the file is created and deleted within the script

File: docker/crypto-detector/consumer/Dockerfile (hunk 6 above)
Comment (jvelazquezm) [reply docker/crypto-detector/consumer/Dockerfile]: Dockerfile now uses python:3.9-slim-buster image

**Review by idomingu — APPROVED** [review]

**Pull request merged**